test-js/test-utils.js

```diff
@@ -0,0 +1,247 @@
+    }
+  });
+  it("should generate a range of numbers, with start and stop values", () => {
+    const numbers = range(10, 13);
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(10);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(11);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(12);
+    }
+  });
+  it("should generate a range of numbers, with start, stop and step values", () => {
+    const numbers = range(10, 15, 2);
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(10);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(12);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(14);
+    }
+  });
+});
+
+describe("chain", () => {
+  it("chain iterators", () => {
+    function* numbers() {
+      yield 1;
+      yield 2;
+    }
+    const chained = chain(numbers(), numbers());
+    expect(chained.next().value).equals(1);
+    expect(chained.next().value).equals(2);
+    expect(chained.next().value).equals(1);
+    expect(chained.next().value).equals(2);
+  });
+});
+
+describe("makeUPlusStringFromCodePoint", () => {
+  it("throw an exception when a not valid parameter given", () => {
+    expect(() => makeUPlusStringFromCodePoint("not-a-number")).to.throw();
+  });
+  it("should not throw an exception for a falsy value", () => {
+    expect(() => makeUPlusStringFromCodePoint("")).to.not.throw();
+  });
+  it("make a number unicode hex", () => {
+    expect(makeUPlusStringFromCodePoint(97)).equals("U+0061"); // a
+    expect(makeUPlusStringFromCodePoint(65)).equals("U+0041"); // A
+  });
+});
+
+describe("getCharFromUnicode", () => {
+  it("should return an empty string if an argument is not passed", () => {
+    expect(getCharFromUnicode()).equals("");
+  });
+  it("should convert a unicode symbol number to a character", () => {
+    expect(getCharFromUnicode(97)).equals("a");
+  });
+});
+
+describe("guessCharFromGlyphName", () => {
+  it("should guess a character from code points in free text", () => {
+    expect(guessCharFromGlyphName("text 0061 text")).equals("a");
+    expect(guessCharFromGlyphName("text ff0061 text")).equals("a");
+    expect(guessCharFromGlyphName("text ff0041 text")).equals("A");
+  });
+});
+
+describe("fileNameExtension", () => {
+  it("should return the file extension of a file name", () => {
+    expect(fileNameExtension("test-utils.js")).equals("js");
+  });
+  it("should work well when there is a dot in the file name", () => {
+    expect(fileNameExtension("test.utils.js")).equals("js");
+  });
+});
+
+describe("arrayExtend", () => {
+  it("should extend arrays with another one", () => {
+    const array = [1, 2, 3];
+    arrayExtend(array, [4, 5]);
+    expect(array).to.eql([1, 2, 3, 4, 5]);
```

Comment:
```suggestion
    expect(array).to.deep.equal([1, 2, 3, 4, 5]);
```

Reply:
Not sure why/how `.eql` works, to be honest.

Reply:
It seems `.eql()` is almost equivalent to `.deep.equal()`, but perhaps not quite. I don't understand the subtlety yet, if there is one.